src/main/scala/tile/RocketTile.scala

```diff
@@ -153,6 +177,32 @@ class RocketTileModuleImp(outer: RocketTile) extends BaseTileModuleImp(outer)
   // reset vector is connected in the Frontend to s2_pc
   core.io.reset_vector := DontCare
 
+  val trace_sink_arbiter = outer.rocketParams.traceParams.map { t =>
+    val arb = Module(new TraceSinkArbiter(outer.traceSinkIds, use_monitor = t.useArbiterMonitor, monitor_name = outer.rocketParams.uniqueName))
+    arb
+  }
+
+  if (outer.rocketParams.traceParams.isDefined) {
+    core.io.trace_core_ingress.get <> outer.trace_encoder.get.module.io.in
+    outer.trace_encoder_controller.foreach { lm =>
+      outer.trace_encoder.get.module.io.control <> lm.module.io.control
+    }
+
```

Comment:
Don't construct the arbiter outside this `if` block... just do it inside.... then you don't need a `Option[Arbiter]`